src/libYARP_math/src/RandScalar.cpp

```diff
@@ -45,12 +44,14 @@ RandScalar::~RandScalar()
 
 double RandScalar::get()
 {
-    return implementation(impl)->random();
+    double ret = implementation(impl)->operator ()();
```

Comment:
A static_cast here would be helpful to make the code more readable.

Reply:
The static cast was already executed by `implementation` function. 
I cleared the code by explicitly declaring the generator https://github.com/robotology/yarp/pull/1444/commits/8ea7a46352a390d2a04241fff0ecd5b6eb7f68c0